File: game-server/src/main/java/work/fking/pangya/game/model/CourseStatistics.java
package work.fking.pangya.game.model;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.game.packet.handler.room.CreateRoomPacketHandler.Course;

public class CourseStatistics {

    private final Course course;

    private CourseStatistics(Course course) {
        this.course = course;
    }

    public static CourseStatistics blank(Course course) {
        return new CourseStatistics(course);
    }

    public void serialize(ByteBuf buffer) {
        buffer.writeByte(course.ordinal()); // courseId
        buffer.writeIntLE(5); // totalStrokes
        buffer.writeIntLE(4); // totalPutts
        buffer.writeIntLE(6); // holesPlayed
        buffer.writeIntLE(1); // total fairway shots
        buffer.writeIntLE(0); // holeInOnes
        buffer.writeIntLE(0);
        buffer.writeIntLE(-6 + course.ordinal()); // totalScore
        buffer.writeByte(-3 + course.ordinal()); // bestScore
        buffer.writeLongLE(654 + course.ordinal() * 70); // bestPangEarned
        buffer.writeIntLE(67108872); // bestScoreWithCharIffId
        buffer.writeByte(0);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/model/CourseStatistics.kt
package work.fking.pangya.game.model

import io.netty.buffer.ByteBuf
import work.fking.pangya.game.room.Course

class CourseStatistics(
    private val course: Course
) {
    fun serialize(buffer: ByteBuf) {
        with(buffer) {
            writeByte(course.ordinal) // courseId
            writeIntLE(5) // totalStrokes
            writeIntLE(4) // totalPutts
            writeIntLE(6) // holesPlayed
            writeIntLE(1) // total fairway shots
            writeIntLE(0) // holeInOnes
            writeIntLE(0)
            writeIntLE(-6 + course.ordinal) // totalScore
            writeByte(-3 + course.ordinal) // bestScore
            writeLongLE((654 + course.ordinal * 70).toLong()) // bestPangEarned
            writeIntLE(67108872) // bestScoreWithCharIffId
            writeByte(0)
        }
    }
}
